### PR TITLE
Support NF4/FP4 data type in weight-only

### PR DESCRIPTION
**Pull request by xin3he**
## Type of Change

feature

## Description

support NF4/FP4 data type in weight-only, allow tuning dtype and compressing nf4/fp4 model

## Expected Behavior & Potential Risk

UT pass

## How has this PR been tested?

local test

## Dependency Change?

N/A

### CODE REVIEW

File: docs/source/quantization_weight_only.md [hunk 1]
```diff
@@ -35,6 +35,7 @@ There are many excellent works for weight only quantization to improve its accur
 ### **Quantization Capability**:
 | Config | Capability |
 | :---: | :---:|
+| dtype | ['int', 'nf4', 'fp4'] |
```
Comment (yiliu30) [inline docs/source/quantization_weight_only.md:38]: Can you add some notes regarding the meanings of `int`, `nf4`, and `fp4`?

**Comment by hshen14** [comment]
Which FP4 format? Should specify FP4_E3M0, FP4_E2M1. You can align with Penghui.

File: neural_compressor/strategy/utils/tuning_space.py [hunk 2]
```diff
@@ -467,6 +478,8 @@ def _parse_cap_helper(self, cap):
                         quant_mode = op_cap['activation']['quant_mode']
                         att_dtype = op_cap[att]['dtype'][0]
                         signed_flag, _data_type = extract_data_type(att_dtype)
+                        if _data_type == 'int':
```
Comment (yiliu30) [inline neural_compressor/strategy/utils/tuning_space.py:481]: The below condition may be more robust.
```python
from neural_compressor.strategy.utils.utility import quant_options
if quant_options.quant_type = 3:  # weight_only
    ...
```

**Comment by xin3he** [comment]
> Which FP4 format? Should specify FP4_E3M0, FP4_E2M1. You can align with Penghui.

FLOAT_MAPPING = {'nf4': NF4, 'fp4': FP4_BNB, 'fp4_bnb': FP4_BNB, 'fp4_e2m1': FP4_E2M1, 'e2m1': FP4_E2M1}
By default, fp4 is the one bitsandbytes used. Developers can select from ['fp4_e2m1',  'fp4_bnb']

**Review by PenghuiCheng — APPROVED** [review]

**Comment by xin3he** [comment]
We still miss GTPQ and TEQ to support NF4/FP4, it will happen in late PRs.

**Pull request merged**